### PR TITLE
cluster-image-registry-operator: run additional tests when platform specific code is changed

### PR DESCRIPTION
**Pull request by dmage**

### CODE REVIEW

File: ci-operator/jobs/openshift/cluster-image-registry-operator/openshift-cluster-image-registry-operator-master-presubmits.yaml [hunk 1]
```diff
@@ -902,8 +907,8 @@ presubmits:
       ci-operator.openshift.io/prowgen-controlled: "true"
       pj-rehearse.openshift.io/can-be-rehearsed: "true"
     name: pull-ci-openshift-cluster-image-registry-operator-master-e2e-vsphere-operator
-    optional: true
```
Comment (adambkaplan) [inline ci-operator/jobs/openshift/cluster-image-registry-operator/openshift-cluster-image-registry-operator-master-presubmits.yaml]: Why drop `optional` here?

File: ci-operator/jobs/openshift/cluster-image-registry-operator/openshift-cluster-image-registry-operator-release-4.5-presubmits.yaml [hunk 2]
```diff
@@ -347,7 +347,7 @@ presubmits:
           secretName: sentry-dsn
     trigger: (?m)^/test( | .* )e2e-aws-upgrade,?($|\s.*)
   - agent: kubernetes
-    always_run: true
+    always_run: false
```
Comment (adambkaplan) [inline ci-operator/jobs/openshift/cluster-image-registry-operator/openshift-cluster-image-registry-operator-release-4.5-presubmits.yaml]: I'd expect there to be a similar change in the `master` job file - did these diverge somehow?

File: ci-operator/jobs/openshift/cluster-image-registry-operator/openshift-cluster-image-registry-operator-master-presubmits.yaml (hunk 1 above)
Comment (dmage) [reply ci-operator/jobs/openshift/cluster-image-registry-operator/openshift-cluster-image-registry-operator-master-presubmits.yaml]: We shouldn't merge PRs that break tests. I have a PR that makes these tests green, once it gets merged, we should keep them green.

If this PR will be merged before https://github.com/openshift/cluster-image-registry-operator/pull/458, we can override e2e-vsphere-operator results.

File: ci-operator/jobs/openshift/cluster-image-registry-operator/openshift-cluster-image-registry-operator-release-4.5-presubmits.yaml (hunk 2 above)
Comment (dmage) [reply ci-operator/jobs/openshift/cluster-image-registry-operator/openshift-cluster-image-registry-operator-release-4.5-presubmits.yaml]: the master job file has `always_run: false`, so I changed the 4.5 job too

File: ci-operator/jobs/openshift/cluster-image-registry-operator/openshift-cluster-image-registry-operator-master-presubmits.yaml (hunk 1 above)
Comment (dmage) [reply ci-operator/jobs/openshift/cluster-image-registry-operator/openshift-cluster-image-registry-operator-master-presubmits.yaml]: Eventually I want to drop all `optional: true` from our jobs.

Comment (adambkaplan) [reply ci-operator/jobs/openshift/cluster-image-registry-operator/openshift-cluster-image-registry-operator-master-presubmits.yaml]: @stevekuznetsov perhaps we need some clarification here.

If a job is marked `optional` but is triggered and then fails (ex: via a `/test` instruction), will that failing test block merge?

Comment (dmage) [reply ci-operator/jobs/openshift/cluster-image-registry-operator/openshift-cluster-image-registry-operator-master-presubmits.yaml]: If it's `optional`, its failures will not block merge.

You can use `always_run: false` if you want to run it only using a `/test` instruction. And `optional: false` if you don't want it to block merge.

Comment (dmage) [reply ci-operator/jobs/openshift/cluster-image-registry-operator/openshift-cluster-image-registry-operator-master-presubmits.yaml]: https://github.com/kubernetes/test-infra/blob/master/prow/jobs.md#requiring-jobs-for-auto-merge-through-tide

**Comment by dmage** [comment]
@adambkaplan ptal

**Pull request merged**